File: src/main/java/com/intellij/tapestry/intellij/toolwindow/nodes/TemplatesNode.java
package com.intellij.tapestry.intellij.toolwindow.nodes;

import com.intellij.tapestry.core.model.presentation.PresentationLibraryElement;
import com.intellij.tapestry.core.resource.IResource;

import javax.swing.tree.DefaultMutableTreeNode;

public class TemplatesNode extends DefaultMutableTreeNode {

    public TemplatesNode(Object userObject) {
        super(userObject);

        for (IResource template : ((PresentationLibraryElement) userObject).getTemplate()) {
            add(new ResourceLeafNode(template));
        }
    }

    public String toString() {
        return "Templates";
    }
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/toolwindow/nodes/TemplatesNode.kt
package com.github.rar91279.plugin.tapestry.intellij.toolwindow.nodes

import com.github.rar91279.plugin.tapestry.core.model.presentation.PresentationLibraryElement
import javax.swing.tree.DefaultMutableTreeNode

class TemplatesNode(element: PresentationLibraryElement) : DefaultMutableTreeNode(element) {

    init {
        for (template in element.template) add(ResourceLeafNode(template))
    }

    override fun toString() = "Templates"
}
